earthfile2llb/converter.go

```diff
@@ -1007,10 +1007,10 @@ func (c *Converter) SaveImage(ctx context.Context, imageNames []string, pushImag
 				shouldPush := pushImages && si.DockerTag != "" && c.opt.DoSaves
 				shouldExportLocally := si.DockerTag != "" && c.opt.DoSaves
 				c.waitBlock().addSaveImage(si, c, shouldPush, shouldExportLocally)
-			} else {
-				c.mts.Final.SaveImages = append(c.mts.Final.SaveImages, si)
-			}
 
+				si.SkipBuilder = true
+			}
+			c.mts.Final.SaveImages = append(c.mts.Final.SaveImages, si)
```

Comment:
This makes sense - there are places in our code where we introspect the `mts` for the saved images (e.g. `WITH DOCKER`), so we should always put them there. It shouldn't mean that we always export them though.